Wangscape/noise/Reseedable.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <memory>
+#include <utility>
+#include <functional>
+#include <noise/noise.h>
+
+namespace noise
+{
+
+typedef module::Module Module;
+typedef std::shared_ptr<Module> ModulePtr;
+typedef std::function<void(int)> SetSeedFn;
+// typedef std::pair<ModulePtr, SetSeedFn> Reseedable;
+struct Reseedable final
+{
+    ModulePtr module;
+    SetSeedFn setSeed;
+
+    double getValue(double x, double y, double z);
+
+    Reseedable abs();
+    Reseedable clamp(double lower, double upper);
+    Reseedable pow(Reseedable& exponent);
+    Reseedable pow(double exponent);
+    Reseedable exp(double base);
+    Reseedable invert();
+    Reseedable scaleBias(double scale, double bias);
+    //Reseedable curve(...?)
+    //Reseedable terrace(...?)
```

Comment:
Is there a need to keep it commented out :wink:? It could be an issue, if you anticipate necessity of implementing these methods.